PayButton/paybutton-server
# Feat/411 4 1 getAddressTransactions grpc refactor

Description:
1st of the sequence of smaller PRs broken from #431. This one has a part of #431 concerning grpc, but it already goes beyond what was built there in two aspects: 
 - it uses Prisma type instead of the new interface `Transfer` 
 - it allows transaction sync to happen in 2 steps: first a small sync of custom length (defined in DEFAULT_NUMBER_OF_TRANSACTIONS_TO_SYNC_INITIALLY, suggested arbitrary value of 200) that brings a configurable amount of transactions to the frontend or whoever requested the sync and after, in the background, the sync continues until all transactions are in the database (this almost solves #403, I see only one improvement left on this front, if we try to save a transaction on the database that is already saved and that address has already been fully synced we should stop, since the past in blockchain is immutable, no point in syncing towards the past if we already have the info from there onwards)
Only grpc was touched in this PR, chronik was not and also tests on transactions and subscribe feature were both commented out as they will be addressed in future PRs.

How to test:
 - clean transactions data on your local db:
    + delete from paybutton.PricesOnTransactions 
    + delete from paybutton.Transaction 
 - tell system to sync any bch address visiting (change address at the end if you wanna test another one): http://localhost:3000/api/address/transactions/sync/bitcoincash:qzqh7rwaq9zm4zcv40lh9c9u50gy07gcesdmja8426 
 - address above has 4 transactions, you can change DEFAULT_NUMBER_OF_TRANSACTIONS_TO_SYNC_INITIALLY to 1 or find an bch address that has more than 200 transactions to test the background sync functionality, repeat test steps above from the beginning after changing this number or finding another bch address, you should see only 1 transaction replied in the webpage (if you changed the value to 1), but in the database you should see all transactions of that address.
